pennylane/labs/resource_estimation/ops/op_math/symbolic.py

```diff
@@ -178,3 +182,114 @@ def pow_resource_decomp(
     def tracking_name(base_class, z, base_params) -> str:
         base_name = base_class.tracking_name(**base_params)
         return f"Pow({base_name}, {z})"
+
+
+class ResourceExp(Exp, re.ResourceOperator):
+    """Resource class for Exp"""
+
+    @staticmethod
+    def _resource_decomp(base_class, coeff, num_steps, **kwargs):
+
+        while isinstance(base_class, SProd):
+            coeff *= base_class.scalar
+            base_class = base_class.base
+
+        # Custom exponential operator resources:
+        if isinstance(base_class, re.ResourceOperator):
+            try:
+                return base_class.exp_resource_decomp(coeff, num_steps, **kwargs)
+            except re.ResourcesNotDefined:
+                pass
+
+        # PauliRot resource decomp:
+        if (pauli_sentence := base_class.pauli_rep) and math.real(coeff) == 0:
```

Comment:
`base_class` should be a type, so either there needs to be a `pauli_rep` attribute in `resource_params` or `pauli_rep` needs to be provided in `base_params` (see my earlier comment about `resource_rep`).